node/lib/util/read_repo_ast_util.js

```diff
@@ -231,17 +231,32 @@ const loadIndexAndWorkdir = co.wrap(function *(repo, headCommit) {
     };
 });
 
+const syntheticRefRegexp = new RegExp("^refs/commits/[0-9a-f]{40}$");
```

Comment:
do we need to worry about capital letters [A-F]?

Reply:
We are responsible for creating these refs, and we create them all-lowercase.